packages/astro/src/i18n/vite-plugin-i18n.ts

```diff
@@ -29,28 +28,16 @@ export default function astroInternationalization({
 	return {
 		name: 'astro:i18n',
 		enforce: 'pre',
-		async resolveId(id) {
+		config(config) {
+			const i18nConfig: I18nInternalConfig = { base, format, site, trailingSlash, i18n };
+			config.define ??= {}
+			config.define.__ASTRO_INTERNAL_I18N_CONFIG__ = JSON.stringify(i18nConfig);
```

Comment:
Tiny nit: I'd do something like:

```js
return {
  define: {
    __ASTRO_INTERNAL_I18N_CONFIG__: ...
  }
}
```

so it's a little cleaner. Though they'd still work the same.

Reply:
That looks cleaner to me as well!

Could you make a suggestion so it can be committed directly?

Reply:
```suggestion
			return {
				define: {
					__ASTRO_INTERNAL_I18N_CONFIG__: JSON.stringify(i18nConfig)
				}
			}
```